src/game/AI/ScriptDevAI/scripts/world/scourge_invasion.cpp

```diff
@@ -362,6 +374,52 @@ struct NecropolisHealthAI : public ScriptedAI
 {
     NecropolisHealthAI(Creature* creature) : ScriptedAI(creature)
     {
+        AddCustomAction(0, 5000u, [&]()
+        {
+            if (m_creature->GetHealthPercent() > 99.f)
+            {
+                CreatureList proxies;
+                GetCreatureListWithEntryInGrid(proxies, m_creature, NPC_NECROPOLIS_PROXY, 200.f);
+
+                for (Creature* proxy : proxies)
+                {
+                    if (proxy && (proxy->IsDespawned() || !proxy->IsAlive()))
+                    {
+                        proxy->SetRespawnTime(0);
+                        proxy->Respawn();
+                    }
+                }
+
+                GameObjectList circles;
+                GetGameObjectListWithEntryInGrid(circles, m_creature, GOBJ_SUMMON_CIRCLE, 1000.f);
```

Comment:
The range of 1000.f is probably too far, I will also need to measure what the distance between the furthest circle is to the necropolis, so we don't accidentally count circles that belong to other Necropoli. This particular part of the code is intended to be run after server restarts when an invasion is resumed. The number of defeated necropoli in an area is saved, but not the number of defeated Necrotic Shards, so if the Health Indicator for a necropolis is at 100% health it means it requires 3 necrotic shards to be defeated, if we spawned any less or some remain despawned because of the restart, this zone would not be defeatable

Reply:
Biggest grid search is 533, and I strongly advise against never using a value higher than 100. This needs to be done differently I am afraid.

Reply:
Do you have any suggestion that would work stateless?
The problem is that these GameObjects aren't summoned, but they're spawned from database, so I have no innate handle on them per necropolis. I can add these objects to a map or list when they spawn, but then I have no real way of associating them with a particular necropolis